src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,414 @@
+#define BMP388_MODE_SLEEP                               (0x00)
+#define BMP388_MODE_FORCED                              (0x01)
+#define BMP388_MODE_NORMAL                              (0x02)
+
+#define BMP388_CALIRATION_LOWER_REG                     (0x30) // See datasheet 4.3.19, "calibration data"
+#define BMP388_TRIMMING_NVM_PAR_T1_LSB_REG              (0x31) // See datasheet 3.11.1 "Memory map trimming coefficients"
+#define BMP388_TRIMMING_NVM_PAR_P11_REG                 (0x45) // See datasheet 3.11.1 "Memory map trimming coefficients"
+#define BMP388_CALIRATION_UPPER_REG                     (0x57)
+
+#define BMP388_TRIMMING_DATA_LENGTH                     ((BMP388_TRIMMING_NVM_PAR_P11_REG - BMP388_TRIMMING_NVM_PAR_T1_LSB_REG) + 1) // +1 for inclusive
+
+#define BMP388_DATA_FRAME_SIZE               (6)
+
+#define BMP388_OVERSAMP_1X               (0x00)
+#define BMP388_OVERSAMP_2X               (0x01)
+#define BMP388_OVERSAMP_4X               (0x02)
+#define BMP388_OVERSAMP_8X               (0x03)
+#define BMP388_OVERSAMP_16X              (0x04)
+#define BMP388_OVERSAMP_32X              (0x05)
+
+// INT_CTRL register
+#define BMP388_INT_OD_BIT                   0
+#define BMP388_INT_LEVEL_BIT                1
+#define BMP388_INT_LATCH_BIT                2
+#define BMP388_INT_FWTM_EN_BIT              3
+#define BMP388_INT_FFULL_EN_BIT             4
+#define BMP388_INT_RESERVED_5_BIT           5
+#define BMP388_INT_DRDY_EN_BIT              6
+#define BMP388_INT_RESERVED_7_BIT           7
+
+// OSR register
+#define BMP388_OSR_P_BIT                    0 // to 2
+#define BMP388_OSR4_T_BIT                   3 // to 5
+#define BMP388_OSR_P_MASK                   (0x03)  // -----111
+#define BMP388_OSR4_T_MASK                  (0x38)  // --111---
+
+// configure pressure and temperature oversampling, forced sampling mode
+#define BMP388_PRESSURE_OSR              (BMP388_OVERSAMP_8X)
+#define BMP388_TEMPERATURE_OSR           (BMP388_OVERSAMP_1X)
+
+// see Datasheet 3.11.1 Memory Map Trimming Coefficients
+typedef struct bmp388_calib_param_s {
+    uint16_t T1;
+    uint16_t T2;
+    int8_t T3;
+    int16_t P1;
+    int16_t P2;
+    int8_t P3;
+    int8_t P4;
+    uint16_t P5;
+    uint16_t P6;
+    int8_t P7;
+    int8_t P8;
+    int16_t P9;
+    int8_t P10;
+    int8_t P11;
+} __attribute__((packed)) bmp388_calib_param_t;
+
+STATIC_ASSERT(sizeof(bmp388_calib_param_t) == BMP388_TRIMMING_DATA_LENGTH, bmp388_calibration_structure_incorrectly_packed);
+
+static uint8_t bmp388_chip_id = 0;
+STATIC_UNIT_TESTED bmp388_calib_param_t bmp388_cal;
+// uncompensated pressure and temperature
+uint32_t bmp388_up = 0;
+uint32_t bmp388_ut = 0;
+STATIC_UNIT_TESTED int64_t t_lin = 0;
+
+static void bmp388_start_ut(baroDev_t *baro);
+static void bmp388_get_ut(baroDev_t *baro);
+static void bmp388_start_up(baroDev_t *baro);
+static void bmp388_get_up(baroDev_t *baro);
+
+STATIC_UNIT_TESTED void bmp388_calculate(int32_t *pressure, int32_t *temperature);
```

Comment:
I do not think that 'legacy code does it as well' is a valid excuse to break the coding style rules with new code. Can this please be changed to comply with the coding style rules?

Reply:
no problem - I will change these and the bmp280 method names in a new commit.

Reply:
done